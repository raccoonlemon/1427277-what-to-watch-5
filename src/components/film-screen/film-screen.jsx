import PropTypes from 'prop-types';
import React, {useEffect} from "react";
import {connect} from 'react-redux';
import {Link} from "react-router-dom";
import {Path} from "../../const";
import {fetchFilmById, fetchReviewsByFilmId} from '../../store/api-actions';
import {selectFilm, selectIsFilmLoaded, selectIsUserLogged, selectReviews, selectSimilarFilms} from '../../store/selectors';
import {filmShape, reviewShape} from "../../utils/props-validation";
import AddToListButton from '../add-to-list-button/add-to-list-button';
import PlayButton from '../play-button/play-button';
import FilmInfo from '../film-info/film-info';
import FilmsList from '../films-list/films-list';
import Footer from '../footer/footer';
import Header from '../header/header';

// TODO: разобраться с ошибкой валидации пропсов
// Warning: Failed prop type: The prop `film.id` is marked as required in `FilmInfo`, but its value is `undefined`.
// Warning: Failed prop type: The prop `isFavorite` is marked as required in `AddToListButton`, but its value is `undefined`.

const FilmScreen = (props) => {

  const {isFilmLoaded, loadFilmInfoAction, id} = props;

  useEffect(() => {
    if (!isFilmLoaded) {
      loadFilmInfoAction(id);
    }
  }, [id]);

  const {film, reviews, similarFilms, isUserLogged} = props;
  const {title, genre, year, poster, background, backgroundColor, isFavorite} = film;
  return (<React.Fragment>
    <section className="movie-card movie-card--full" style={{backgroundColor}}>
      <div className="movie-card__hero">
        <div className="movie-card__bg">
          <img src={background} alt={title} />
        </div>

        <h1 className="visually-hidden">WTW</h1>
        <Header className="movie-card__head"/>

        <div className="movie-card__wrap">
          <div className="movie-card__desc">
            <h2 className="movie-card__title">{title}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{genre}</span>
              <span className="movie-card__year">{year}</span>
            </p>

            <div className="movie-card__buttons">
              <PlayButton id = {id}></PlayButton>
              <AddToListButton id = {id} isFavorite = {isFavorite}/>
              {isUserLogged && <Link className="btn movie-card__button" to={Path.addReview(id)}>Add review</Link>}
            </div>
          </div>
        </div>
      </div>

      <div className="movie-card__wrap movie-card__translate-top">
        <div className="movie-card__info">
          <div className="movie-card__poster movie-card__poster--big">
            <img src={poster} alt={`${title} poster`} width="218" height="327" />
          </div>
          <FilmInfo film={film} reviews={reviews}/>
        </div>
      </div>
    </section>

    <div className="page-content">

      {similarFilms.length > 0 &&
        <section className="catalog catalog--like-this">
          <h2 className="catalog__title">More like this</h2>
          <FilmsList films = {similarFilms}/>
        </section>}

      <Footer/>

    </div>
  </React.Fragment>);
};

FilmScreen.propTypes = {
  loadFilmInfoAction: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  film: filmShape.isRequired,
  isFilmLoaded: PropTypes.bool.isRequired,
  reviews: PropTypes.arrayOf(reviewShape).isRequired,
  similarFilms: PropTypes.arrayOf(filmShape).isRequired,
  isUserLogged: PropTypes.bool.isRequired,
};

const mapStateToProps = (state, {id}) => ({
  film: selectFilm(state),
  isFilmLoaded: selectIsFilmLoaded(id)(state),
  reviews: selectReviews(state),
  similarFilms: selectSimilarFilms(state),
  isUserLogged: selectIsUserLogged(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadFilmInfoAction(id) {
    dispatch(fetchFilmById(id));
    dispatch(fetchReviewsByFilmId(id));
  },
});

export {FilmScreen};
export default connect(mapStateToProps, mapDispatchToProps)(FilmScreen);

