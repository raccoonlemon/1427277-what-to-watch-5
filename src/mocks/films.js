export const films = [
  {
    id: `1`,
    title: `The Grand Budapest Hotel`,
    genre: `drama`,
    year: 2014,
    director: `Wes Andreson`,
    duration: 124,
    starring:
    [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterSmall: `img/the-grand-budapest-hotel-poster.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `2`,
    title: `Pulp fiction`,
    genre: `crime`,
    year: 1994,
    director: `Quentin Tarantino`,
    duration: 118,
    starring:
    [
      `Tim Roth`,
      `Amanda Plummer`,
      `Laura Lovelace`,
      `John Travolta`,
      `Samuel L. Jackson`,
      `Phil LaMarr`,
      `Frank Whaley`,
      `Burr Steers`,
      `Bruce Willis`
    ],
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    background: `img/pulp-fiction.jpg`,
    poster: `img/pulp-fiction.jpg`,
    posterSmall: `img/pulp-fiction.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `3`,
    title: `Moonrise Kingdom`,
    genre: `romance`,
    year: 2012,
    director: `Wes Anderson`,
    duration: 98,
    starring:
    [
      `Bruce Willis`,
      `Edward Norton`,
      `Bill Murray`,
      `Frances McDormand`,
      `Tilda Swinton`,
      `Jared Gilman`,
      `Kara Hayward`,
      `Jason Schwartzman`,
      `Bob Balaban`
    ],
    description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
    background: `img/moonrise-kingdom.jpg`,
    poster: `img/moonrise-kingdom.jpg`,
    posterSmall: `img/moonrise-kingdom.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `4`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `fantasy`,
    year: 2018,
    director: `David Yates`,
    duration: 134,
    starring:
    [
      `Johnny Depp`,
      `Kevin Guthrie`,
      `Carmen Ejogo`,
      `Wolf Roth`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
      `Cornell John`
    ],
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterSmall: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `5`,
    title: `Aviator`,
    genre: `biography`,
    year: 2004,
    director: `Martin Scorsese`,
    duration: 110,
    starring:
    [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`,
      `John C. Reilly`,
      `Alec Baldwin`,
      `Alan Alda`,
      `Jude Law`,
    ],
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    background: `img/aviator.jpg`,
    poster: `img/aviator.jpg`,
    posterSmall: `img/aviator.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: `6`,
    title: `Shutter Island`,
    genre: `thriller `,
    year: 2010,
    director: `Martin Scorsese`,
    duration: 142,
    starring:
    [
      `Leonardo DiCaprio`,
      `Mark Ruffalo`,
      `Ben Kingsley`,
      `Michelle Williams`,
      `Emily Mortimer`,
      `Patricia Clarkson`,
      `Max von Sydow`,
    ],
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    background: `img/shutter-island.jpg`,
    poster: `img/shutter-island.jpg`,
    posterSmall: `img/shutter-island.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `7`,
    title: `Johnny English`,
    genre: `comedy `,
    year: 2003,
    director: `Peter Howitt`,
    duration: 109,
    starring:
    [
      `Rowan Atkinson`,
      `Natalie Imbruglia`,
      `Ben Miller`,
      `John Malkovich`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
    ],
    description: `After a cyber-attack reveals the identity of all of the active undercover agents in Britain, Johnny English (Rowan Atkinson) is forced to come out of retirement to find the mastermind hacker.`,
    background: `img/johnny-english.jpg`,
    poster: `img/johnny-english.jpg`,
    posterSmall: `img/johnny-english.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `8`,
    title: `Seven Years in Tibet`,
    genre: `drama`,
    year: 1997,
    director: `Jean-Jacques Annaud`,
    duration: 125,
    starring:
    [
      `Brad Pitt`,
      `David Thewlis`,
      `BD Wong`,
      `Danny Denzongpa`,
      `Mako`,
      `Jamyang Jamtsho Wangchuk`,
      `Lhakpa Tsamchoe`,
      `Jetsun Pema`,
    ],
    description: `True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.`,
    background: `img/seven-years-in-tibet.jpg`,
    poster: `img/seven-years-in-tibet.jpg`,
    posterSmall: `img/seven-years-in-tibet.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
];

export const promoFilm = films[0];
