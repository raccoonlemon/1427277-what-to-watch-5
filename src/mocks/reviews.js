const texts = [
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
  `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.`,
  `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`
];

export const reviews = [
  {
    id: 1,
    date: new Date(2020, 10, 10, 0, 0, 0, 0),
    author: `Ellie`,
    rating: 10,
    text: texts[0],
  },
  {
    id: 2,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Max`,
    rating: 5,
    text: texts[1]
  },
  {
    id: 3,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Parvati`,
    rating: 8,
    text: texts[2]
  },
  {
    id: 4,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Felix`,
    rating: 8,
    text: texts[3]
  },
  {
    id: 5,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Nyoka`,
    rating: 4,
    text: texts[4]
  }
];
