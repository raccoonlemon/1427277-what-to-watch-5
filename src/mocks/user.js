export const user = {
  id: `1`,
  email: `email@email.com`,
  name: `User Name`,
  avatarSrc: `src`,
};
