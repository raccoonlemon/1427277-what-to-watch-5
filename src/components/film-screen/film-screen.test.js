import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {film, films} from "../../mocks/films";
import {reviews} from "../../mocks/reviews";
import {FilmScreen} from "./film-screen";

const renderer = new ShallowRenderer();

const noop = ()=>{};
it(`<FilmScreen> renders correctly`, () => {
  renderer.render(
      <FilmScreen
        id = "1"
        film = {film}
        isFilmLoaded = {true}
        reviews = {reviews}
        loadFilmInfoAction={noop}
        similarFilms = {films}
        isUserLogged = {false}
      />
  );
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
