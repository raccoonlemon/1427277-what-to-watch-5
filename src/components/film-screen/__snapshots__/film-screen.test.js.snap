// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmScreen> renders correctly 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#eee5b5",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Header
        className="movie-card__head"
        needToShowUserMenu={true}
      />
      <div
        className="movie-card__wrap"
      >
        <FilmCard
          film={
            Object {
              "background": "img/bg-the-grand-budapest-hotel.jpg",
              "backgroundColor": "#eee5b5",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "duration": 124,
              "genre": Array [
                "crime",
                "comedy",
              ],
              "id": "1",
              "isFavorite": true,
              "poster": "img/the-grand-budapest-hotel-poster.jpg",
              "posterSmall": "img/the-grand-budapest-hotel-poster.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
                "Tony Revoloru",
                "Tom Wilkinson",
                "Owen Wilkinson",
                "Adrien Brody",
                "Ralph Fiennes",
                "Jeff Goldblum",
              ],
              "title": "The Grand Budapest Hotel",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2014,
            }
          }
          isUserLogged={false}
        />
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <FilmInfo
          film={
            Object {
              "background": "img/bg-the-grand-budapest-hotel.jpg",
              "backgroundColor": "#eee5b5",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "duration": 124,
              "genre": Array [
                "crime",
                "comedy",
              ],
              "id": "1",
              "isFavorite": true,
              "poster": "img/the-grand-budapest-hotel-poster.jpg",
              "posterSmall": "img/the-grand-budapest-hotel-poster.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
                "Tony Revoloru",
                "Tom Wilkinson",
                "Owen Wilkinson",
                "Adrien Brody",
                "Ralph Fiennes",
                "Jeff Goldblum",
              ],
              "title": "The Grand Budapest Hotel",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2014,
            }
          }
          reviews={
            Array [
              Object {
                "author": "Ellie",
                "date": 1920-11-09T18:52:12.000Z,
                "id": 1,
                "rating": 10,
                "text": "It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.",
              },
              Object {
                "author": "Max",
                "date": 1920-11-11T18:52:12.000Z,
                "id": 2,
                "rating": 5,
                "text": "Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.",
              },
              Object {
                "author": "Parvati",
                "date": 1920-11-11T18:52:12.000Z,
                "id": 3,
                "rating": 8,
                "text": "I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.",
              },
              Object {
                "author": "Felix",
                "date": 1920-11-11T18:52:12.000Z,
                "id": 4,
                "rating": 8,
                "text": "The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.",
              },
              Object {
                "author": "Nyoka",
                "date": 1920-11-11T18:52:12.000Z,
                "id": 5,
                "rating": 4,
                "text": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
              },
            ]
          }
        />
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <FilmsList
        films={
          Array [
            Object {
              "background": "img/bg-the-grand-budapest-hotel.jpg",
              "backgroundColor": "#eee5b5",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "duration": 124,
              "genre": Array [
                "crime",
                "comedy",
              ],
              "id": "1",
              "isFavorite": true,
              "poster": "img/the-grand-budapest-hotel-poster.jpg",
              "posterSmall": "img/the-grand-budapest-hotel-poster.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
                "Tony Revoloru",
                "Tom Wilkinson",
                "Owen Wilkinson",
                "Adrien Brody",
                "Ralph Fiennes",
                "Jeff Goldblum",
              ],
              "title": "The Grand Budapest Hotel",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2014,
            },
            Object {
              "background": "img/pulp-fiction.jpg",
              "backgroundColor": "#eee5b5",
              "description": "The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 118,
              "genre": "crime",
              "id": "2",
              "isFavorite": false,
              "poster": "img/pulp-fiction.jpg",
              "posterSmall": "img/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Tim Roth",
                "Amanda Plummer",
                "Laura Lovelace",
                "John Travolta",
                "Samuel L. Jackson",
                "Phil LaMarr",
                "Frank Whaley",
                "Burr Steers",
                "Bruce Willis",
              ],
              "title": "Pulp fiction",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 1994,
            },
            Object {
              "background": "img/moonrise-kingdom.jpg",
              "backgroundColor": "#eee5b5",
              "description": "A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.",
              "director": "Wes Anderson",
              "duration": 98,
              "genre": Array [
                "comedy",
                "drama",
              ],
              "id": "3",
              "isFavorite": false,
              "poster": "img/moonrise-kingdom.jpg",
              "posterSmall": "img/moonrise-kingdom.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Bruce Willis",
                "Edward Norton",
                "Bill Murray",
                "Frances McDormand",
                "Tilda Swinton",
                "Jared Gilman",
                "Kara Hayward",
                "Jason Schwartzman",
                "Bob Balaban",
              ],
              "title": "Moonrise Kingdom",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2012,
            },
            Object {
              "background": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "backgroundColor": "#e1b0b2",
              "description": "The second installment of the \\"Fantastic Beasts\\" series featuring the adventures of Magizoologist Newt Scamander.",
              "director": "David Yates",
              "duration": 134,
              "genre": Array [
                "fantasy",
                "adventure",
              ],
              "id": "4",
              "isFavorite": false,
              "poster": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "posterSmall": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Johnny Depp",
                "Kevin Guthrie",
                "Carmen Ejogo",
                "Wolf Roth",
                "Eddie Redmayne",
                "Zoë Kravitz",
                "Callum Turner",
                "Derek Riddell",
                "Cornell John",
              ],
              "title": "Fantastic Beasts: The Crimes of Grindelwald",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2018,
            },
            Object {
              "background": "img/aviator.jpg",
              "backgroundColor": "#e1b0b2",
              "description": "A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.",
              "director": "Martin Scorsese",
              "duration": 110,
              "genre": Array [
                "drama",
                "biography",
              ],
              "id": "5",
              "isFavorite": true,
              "poster": "img/aviator.jpg",
              "posterSmall": "img/aviator.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Leonardo DiCaprio",
                "Cate Blanchett",
                "Kate Beckinsale",
                "John C. Reilly",
                "Alec Baldwin",
                "Alan Alda",
                "Jude Law",
              ],
              "title": "Aviator",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2004,
            },
            Object {
              "background": "img/shutter-island.jpg",
              "backgroundColor": "#e1b0b2",
              "description": "In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.",
              "director": "Martin Scorsese",
              "duration": 142,
              "genre": "thriller",
              "id": "6",
              "isFavorite": false,
              "poster": "img/shutter-island.jpg",
              "posterSmall": "img/shutter-island.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Leonardo DiCaprio",
                "Mark Ruffalo",
                "Ben Kingsley",
                "Michelle Williams",
                "Emily Mortimer",
                "Patricia Clarkson",
                "Max von Sydow",
              ],
              "title": "Shutter Island",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2010,
            },
            Object {
              "background": "img/johnny-english.jpg",
              "backgroundColor": "#e1b0b2",
              "description": "After a cyber-attack reveals the identity of all of the active undercover agents in Britain, Johnny English (Rowan Atkinson) is forced to come out of retirement to find the mastermind hacker.",
              "director": "Peter Howitt",
              "duration": 109,
              "genre": Array [
                "comedy",
                "adventure",
              ],
              "id": "7",
              "isFavorite": true,
              "poster": "img/johnny-english.jpg",
              "posterSmall": "img/johnny-english.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Rowan Atkinson",
                "Natalie Imbruglia",
                "Ben Miller",
                "John Malkovich",
                "Eddie Redmayne",
                "Zoë Kravitz",
                "Callum Turner",
                "Derek Riddell",
              ],
              "title": "Johnny English",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 2003,
            },
            Object {
              "background": "img/seven-years-in-tibet.jpg",
              "backgroundColor": "#e1b0b2",
              "description": "True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.",
              "director": "Jean-Jacques Annaud",
              "duration": 125,
              "genre": "drama",
              "id": "8",
              "isFavorite": false,
              "poster": "img/seven-years-in-tibet.jpg",
              "posterSmall": "img/seven-years-in-tibet.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "starring": Array [
                "Brad Pitt",
                "David Thewlis",
                "BD Wong",
                "Danny Denzongpa",
                "Mako",
                "Jamyang Jamtsho Wangchuk",
                "Lhakpa Tsamchoe",
                "Jetsun Pema",
              ],
              "title": "Seven Years in Tibet",
              "videoSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "year": 1997,
            },
          ]
        }
      />
    </section>
    <Footer
      className=""
    />
  </div>
</React.Fragment>
`;
